packages/matter.js/src/cluster/definitions/GeneralCommissioningCluster.ts

```diff
@@ -0,0 +1,437 @@
+     * safe timer can be re-armed. See Section 11.9.6.2.1, “Fail Safe Context”.
+     *
+     * The value of this field shall be greater than or equal to the FailSafeExpiryLengthSeconds. Absent additional
+     * guidelines, it is RECOMMENDED that the value of this field be aligned with Section 5.4.2.3, “Announcement
+     * Duration” and default to 900 seconds.
+     *
+     * @see {@link MatterCoreSpecificationV1_1} § 11.9.4.3.2
+     */
+    maxCumulativeFailsafeSeconds: TlvField(1, TlvUInt16)
+});
+
+/**
+ * This enumeration is used by the RegulatoryConfig and LocationCapability attributes to indicate possible radio usage.
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.4.2
+ */
+export const enum RegulatoryLocationType {
+    Indoor = 0,
+    Outdoor = 1,
+    IndoorOutdoor = 2
+}
+
+/**
+ * Input to the GeneralCommissioning armFailSafe command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.2
+ */
+export const TlvArmFailSafeRequest = TlvObject({
+    expiryLengthSeconds: TlvField(0, TlvUInt16),
+    breadcrumb: TlvField(1, TlvUInt64)
+});
+
+/**
+ * This enumeration is used by several response commands in this cluster to indicate particular errors.
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.4.1
+ */
+export const enum CommissioningError {
+    Ok = 0,
+    ValueOutsideRange = 1,
+    InvalidAuthentication = 2,
+    NoFailSafe = 3,
+    BusyWithOtherAdmin = 4
+}
+
+/**
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.3
+ */
+export const TlvArmFailSafeResponse = TlvObject({
+    /**
+     * This field shall contain the result of the operation, based on the behavior specified in the functional
+     * description of the ArmFailSafe command.
+     *
+     * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.3.1
+     */
+    errorCode: TlvField(0, TlvEnum<CommissioningError>()),
+
+    /**
+     * See Section 11.9.6.1, “Common fields in General Commissioning cluster responses”.
+     *
+     * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.3.2
+     */
+    debugText: TlvField(1, TlvString.bound({ maxLength: 128 }))
+});
+
+/**
+ * Input to the GeneralCommissioning setRegulatoryConfig command
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.4
+ */
+export const TlvSetRegulatoryConfigRequest = TlvObject({
+    newRegulatoryConfig: TlvField(0, TlvEnum<RegulatoryLocationType>()),
+    countryCode: TlvField(1, TlvString.bound({ minLength: 2, maxLength: 2 })),
+    breadcrumb: TlvField(2, TlvUInt64)
+});
+
+/**
+ * This field shall contain the result of the operation, based on the behavior specified in the functional description
+ * of the SetRegulatoryConfig command.
+ *
+ * See Section 11.9.6.1, “Common fields in General Commissioning cluster responses”.
+ *
+ * @see {@link MatterCoreSpecificationV1_1} § 11.9.6.5
+ */
+export const TlvSetRegulatoryConfigResponse = TlvObject({
+    errorCode: TlvField(0, TlvEnum<CommissioningError>()),
+    debugText: TlvField(1, TlvString)
```

Comment:
This was in the spec so no override necessary after previous fixes

Reply:
but 128 maxlength still missing?

Reply:
Hmm yes, so I was looking at ArmFailSafeResponse.debugText...  It's the one that's constrained to 128.  For SetRegulatoryConfigResponse and CommissioningCompleteResponse the spec has no constraint.  Maybe they did this on purpose under the theory that failsafe timeout may be due to low-memory condition?

We used to constraint to max 128 for all three but that was probably because we used the same structure for all of them.  I'm inclined to leave as defined by spec, WDYT?

FWIW CHIP doesn't constrain any of the three.